psycopg/psycopgmodule.c

```diff
@@ -109,12 +189,16 @@ psyco_connect(PyObject *self, PyObject *args, PyObject *keywds)
         conn = PyObject_CallFunction(factory, "si", dsn, async);
```

Comment:
And actually, what would stop us from passing the key-value pairs straight to `PQconnectdbParams()` which is available since 9.0?

The `dsn` then will have to become a calculated property of course, so if requested we could still create a conninfo string, to use in `repr()`, for example, but if the user never asks for it, why bother making one?